admiral/pkg/clusters/registry.go

```diff
@@ -83,6 +88,24 @@ func InitAdmiral(ctx context.Context, params common.AdmiralParams) (*RemoteRegis
 	return &w, nil
 }
 
+func pauseForAdmiralToInitializeState(){
+	// Sleep until Admiral determines state. This is done to make sure events are not skipped during startup while determining READ-WRITE state
+	start := time.Now()
+	log.Info("Pausing thread to let Admiral determine it's READ-WRITE state. This is to let Admiral determine it's state during startup")
+	for {
+		if CurrentAdmiralState.IsStateInitialized {
+			log.Infof("Time taken for Admiral to complete state initialization =%v ms", time.Since(start).Milliseconds())
+			break
+		}
+		if time.Since(start).Milliseconds() > 60000 {
```

Comment:
Any reason behind 60 seconds? Should this be based on lock acquire time?

Reply:
Lock acquire time is approx 150-200ms. I did not want to keep the threshold too low to avoid crashing the pod prematurely